app/src/components/map/MapSelectLocation.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
+import { Dimensions, StyleSheet, View } from 'react-native';
+import MapView, { LatLng, MapEvent, Marker } from 'react-native-maps';
+
+interface MapSelectLocationProps {
+  onMarkerPlaced: (arg: LatLng) => void;
+}
+
+export const MapSelectLocation: React.FC<MapSelectLocationProps> = (props) => {
+  const [markerC, setMarkerC] = useState<LatLng>();
+
+  const handlePress = (event: MapEvent) => {
+    setMarkerC(event.nativeEvent.coordinate);
+    props.onMarkerPlaced({
+      latitude: event.nativeEvent.coordinate.latitude,
+      longitude: event.nativeEvent.coordinate.longitude,
+    });
```

Comment:
`event.nativeEvent.coordinate` zwraca obiekt typu `LatLng`, więc możesz bezpośrednio przekazać go do `onMarkerPlaced`
```suggestion
    props.onMarkerPlaced(event.nativeEvent.coordinate);
```